src/plugins/chat/public/components/chat_window.tsx

```diff
@@ -106,6 +106,19 @@ const ChatWindowContent = React.forwardRef<ChatWindowInstance, ChatWindowProps>(
     };
   }, [eventHandler]);
 
+  // Restore timeline from current chat state on component mount
+  useEffect(() => {
+    const currentMessages = chatService.getCurrentMessages();
+    if (currentMessages.length > 0) {
+      setTimeline(currentMessages);
+    }
+  }, [chatService]);
+
+  // Sync timeline changes with ChatService for persistence
+  useEffect(() => {
```

Comment:
Nit: What about change to `useUpdateEffect` here? For now it seems required two useEffect execution in order. If this `updateCurrentMessages` execution before `getCurrentMessages` above when component mount. The timeline in session storage will be cleared.